OpenRA.Mods.Cnc/Traits/SupportPowers/GrantPrerequisiteChargeDrainPower.cs

```diff
@@ -0,0 +1,211 @@
+		public readonly string ActiveText = "ACTIVE";
+
+		[Translate]
+		[Desc("Label to display over the support power icon and in its tooltip while the power is available but not active.")]
+		public readonly string AvailableText = "READY";
+
+		IEnumerable<string> ITechTreePrerequisiteInfo.Prerequisites(ActorInfo info)
+		{
+			yield return Prerequisite;
+		}
+
+		public override object Create(ActorInitializer init) { return new GrantPrerequisiteChargeDrainPower(init.Self, this); }
+	}
+
+	public class GrantPrerequisiteChargeDrainPower : SupportPower, ITechTreePrerequisite, INotifyOwnerChanged
+	{
+		readonly GrantPrerequisiteChargeDrainPowerInfo info;
+		TechTree techTree;
+		bool active;
+
+		public GrantPrerequisiteChargeDrainPower(Actor self, GrantPrerequisiteChargeDrainPowerInfo info)
+			: base(self, info)
+		{
+			this.info = info;
+		}
+
+		protected override void Created(Actor self)
+		{
+			// Special case handling is required for the Player actor.
+			// Created is called before Player.PlayerActor is assigned,
+			// so we must query other player traits from self, knowing that
+			// it refers to the same actor as self.Owner.PlayerActor
+			var playerActor = self.Info.Name == "player" ? self : self.Owner.PlayerActor;
+
+			techTree = playerActor.Trait<TechTree>();
+
+			base.Created(self);
+		}
+
+		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
+		{
+			techTree = newOwner.PlayerActor.Trait<TechTree>();
+			active = false;
+		}
+
+		public override SupportPowerInstance CreateInstance(string key, SupportPowerManager manager)
+		{
+			return new DischargableSupportPowerInstance(key, info, manager);
+		}
+
+		public void Activate(Actor self, SupportPowerInstance instance)
+		{
+			active = true;
+			techTree.ActorChanged(self);
+		}
+
+		public void Deactivate(Actor self, SupportPowerInstance instance)
+		{
+			active = false;
+			techTree.ActorChanged(self);
+		}
+
+		IEnumerable<string> ITechTreePrerequisite.ProvidesPrerequisites
+		{
+			get
+			{
+				if (!active)
+					yield break;
+
+				yield return info.Prerequisite;
+			}
+		}
+
+		public class DischargableSupportPowerInstance : SupportPowerInstance
+		{
+			// Whether the power is available to activate (even if not fully charged)
+			bool available;
+
+			// Whether the power is active right now
+			// Note that this is fundamentally different to SupportPowerInstance.Active
+			// which has a much closer meaning to available above.
+			bool active;
+
+			// Additional discharge rate accrued from damage
+			int additionalDischargeSubTicks = 0;
+
+			public DischargableSupportPowerInstance(string key, GrantPrerequisiteChargeDrainPowerInfo info, SupportPowerManager manager)
+				: base(key, info, manager) { }
+
+			void Deactivate()
+			{
+				active = false;
+				notifiedCharging = false;
+
+				// Fully depleting the charge disables the power until it is again fully charged
+				if (!Active || remainingSubTicks >= TotalTicks * 100)
```

Comment:
`TotalTicks * 100`
Not sure if its worth to extract this to a property and use it here and the two places below.

Reply:
IMO not worth it.